### PR TITLE
OCPBUGS-24486: Enforce the use of shared health checks

### PR DESCRIPTION
**Pull request by nrb**
Support for shared health checks is now [accepted upstream](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/4891), and OpenShift would like to adopt it over the default service node ports.

### CODE REVIEW

File: pkg/cloud/azure/azure.go [hunk 1]
```diff
@@ -173,6 +173,10 @@ func CloudConfigTransformer(source string, infra *configv1.Infrastructure, netwo
 		cfg.VMType = azureconsts.VMTypeStandard
 	}
 
+	// Ensure we are checking each service node port when making health checks
+	// on service load balancers.
+	cfg.ClusterServiceLoadBalancerHealthProbeMode = azureconsts.ClusterServiceLoadBalancerHealthProbeModeServiceNodePort
```
Comment (JoelSpeed) [inline pkg/cloud/azure/azure.go:178]: This should be the shared option, not the node port. Node port is the old broken behaviour

Comment (nrb) [reply pkg/cloud/azure/azure.go]: Fixed.

**Comment by nrb** [comment]
/hold

This needs to wait for https://github.com/openshift/cloud-provider-azure/pull/88 to merge in order for the Azure imports to work correctly.

**Comment by damdo** [comment]
@nrb title for this PR will need changing :)

**Comment by nrb** [comment]
@damdo Fixed title and description, thanks

**Comment by nrb** [comment]
This latest push should satisfy building, though some unit tests are still not passing.

File: pkg/controllers/cloud_config_sync_controller_test.go [hunk 2]
```diff
@@ -264,30 +264,26 @@ var _ = Describe("Cloud config sync controller", func() {
 	})
 
 	It("config should be synced up after first reconcile", func() {
-		Eventually(func() (bool, error) {
+		Eventually(func(g Gomega) (string, error) {
```
[the remark this answers is not in the document]
Comment (nrb) [reply pkg/controllers/cloud_config_sync_controller_test.go]: Fixed.

**Comment by damdo** [comment]
Has the images build test OOM'd?

https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-cloud-controller-manager-operator/326/pull-ci-openshift-cluster-cloud-controller-manager-operator-master-images/1750549543109267456#1:build-log.txt%3A70

Should we bump its memory resources?

**Comment by JoelSpeed** [comment]
/test images

Lets just check again but yes, looks like an oom

**Pull request merged**